blazar/plugins/networks/network_plugin.py

```diff
@@ -634,6 +636,59 @@ def delete_network(self, network_id):
             # they have to rerun
             raise manager_ex.CantDeleteNetwork(network=network_id, msg=str(e))
 
+    def list_allocations(self, query, detail=False):
+        network_id_list = [n['id'] for n in db_api.network_list()]
+        options = self.get_query_options(query, QUERY_TYPE_ALLOCATION)
+        options['detail'] = detail
+
+        network_allocations = self.query_network_allocations(network_id_list,
+                                                             **options)
+        return [{"resource_id": network, "reservations": allocs}
```

Comment:
How is the resource_id actually defined for a network? The network_id doesn't seem sufficient, as it is not unique... there are many different resources in Blazar for a given network. The appropriate id seems to be a combination of network_id and segment_id, like `<network_uuid>:<segment_id>` perhaps.